specifyweb/frontend/js_src/lib/components/FormEditor/fetchAllViews.ts

```diff
@@ -91,6 +102,8 @@ const augmentDiskView = (view: ViewDefinition): PresentableViewDefinition => ({
     typeof view.viewsetFile === 'string'
       ? filePathToHuman(view.viewsetFile)
       : camelToHuman(view.viewsetLevel),
+  disciplineId: undefined,
```

Comment:
I might be misunderstanding, but what is the trouble with getting the disciplineId in here just like you did it in the place above?

i.e
```ts
  disciplineId: strictParseResourceUrl(
    defined(
      userInformation.availableCollections.find(
        ({ id }) => id === view.collectionId
      )?.discipline
    )
  ).at(1) as number
```

> I'm not sure this worst the time as the aim is to get rid of this disk files.

or do you also need to do some react code changes to accommodate this (in addition to the changes you already made)?